File: test_commits/python/bug_fix_changed5.py
class StatsCalculatorChanged:
    def __init__(self, numbers):
        if numbers is None:
            raise ValueError("Input must not be None")
        self.numbers = numbers

    def get_average(self):
        if not self.numbers:
            raise ValueError("Cannot calculate average of an empty list")
        total = sum(self.numbers)
        return total / len(self.numbers)

    def get_median(self):
        if not self.numbers:
            raise ValueError("Cannot calculate median of an empty list")
        sorted_nums = sorted(self.numbers)
        n = len(sorted_nums)
        mid = n // 2
        if n % 2 == 0:
            return (sorted_nums[mid - 1] + sorted_nums[mid]) / 2
        return sorted_nums[mid]


if __name__ == "__main__":
    data1 = [10, 20, 30, 40]
    stats1 = StatsCalculatorChanged(data1)
    print(stats1.get_average())
    print(stats1.get_median())

    data2 = []
    stats2 = StatsCalculatorChanged(data2)
    try:
        print(stats2.get_average())
    except Exception as e:
        print(e)

    try:
        print(stats2.get_median())
    except Exception as e:
        print(e)

File: test_commits/javascript/bug_fix_changed5.js
class StatsCalculator {
    constructor(numbers) {
        if (!Array.isArray(numbers)) {
            throw new TypeError("Input must be an array");
        }
        this.numbers = numbers;
    }

    getAverage() {
        if (this.numbers.length === 0) {
            throw new Error("Cannot calculate average of an empty array");
        }
        const sum = this.numbers.reduce((a, b) => a + b, 0);
        return sum / this.numbers.length;
    }

    getMedian() {
        if (this.numbers.length === 0) {
            throw new Error("Cannot calculate median of an empty array");
        }
        const sorted = [...this.numbers].sort((a, b) => a - b);
        const mid = Math.floor(sorted.length / 2);
        if (sorted.length % 2 === 0) {
            return (sorted[mid - 1] + sorted[mid]) / 2;
        }
        return sorted[mid];
    }
}

const data1 = [10, 20, 30, 40];
const stats1 = new StatsCalculator(data1);
console.log(stats1.getAverage());
console.log(stats1.getMedian());

const data2 = [];
const stats2 = new StatsCalculator(data2);
try {
    console.log(stats2.getAverage());
} catch (e) {
    console.error(e.message);
}
try {
    console.log(stats2.getMedian());
} catch (e) {
    console.error(e.message);
}
